File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
int lengthOfLongestSubstring(string s)
{
    if( s.size() ==0 )
        return 0 ;
    set<char> st;
    int mx= 0;
    for( int i =0 ;i< s.size() ;i++ )
    {
        for( int j=i ;j< s.size() ;j++ )
            if(st.count(s[j])==0)
            {
                st.insert(s[j]);
                mx= max((int)st.size() , mx);
            }
        else
            {
            st.clear();
            break;
            }
    }
    return mx;

}
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  if (s.length == 0) return 0;
  else if (s.length == 1 || (s.length == 2 && s[0] == s[1])) return 1;

  let hash = new Map();
  let result = 0;
  let current = 0;

  for (let i = 0; i < s.length; i++) {
    for (let j = i; j < s.length; j++) {
      if (hash.get(s[j]) == 1) {
        result = Math.max(result, current);
        current = 0;
        hash.clear();
        break;
      } else if (!hash.get(s[j])) {
        hash.set(s[j], 1);
        current++;
        result = Math.max(result, current);
      }
    }
  }
  return result;
};